adagrams/game.py

```diff
@@ -1,11 +1,149 @@
-def draw_letters():
-    pass
+import random
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+SCORE_CHART = {
+    'A': 1, 
+    'B': 3, 
+    'C': 3, 
+    'D': 2, 
+    'E': 1, 
+    'F': 4, 
+    'G': 2, 
+    'H': 4, 
+    'I': 1, 
+    'J': 8, 
+    'K': 5, 
+    'L': 1, 
+    'M': 3, 
+    'N': 1, 
+    'O': 1, 
+    'P': 3, 
+    'Q': 10, 
+    'R': 1, 
+    'S': 1, 
+    'T': 1, 
+    'U': 1, 
+    'V': 4, 
+    'W': 4, 
+    'X': 8, 
+    'Y': 4, 
+    'Z': 10
+}
 
+def draw_letters():
+    letter_freq = {}
+    random_letters=[]
+    selected_letters =[]
+    while len(random_letters) < 26:
+        r = random.choice(list(LETTER_POOL.keys()))
+        random_letters.append(r)
+    for letter in random_letters:
+        if letter in letter_freq:
+            letter_freq[letter] += 1
+        else:
+            letter_freq[letter] = 1
+        if letter_freq[letter] <= LETTER_POOL[letter]:
+            selected_letters.append(letter)
+    return (selected_letters[0:10])
+    
 def uses_available_letters(word, letter_bank):
-    pass
+    word_str = ""
+    temp_word = ""
+    letter_bank_dic = {}
+    word_dic ={}
+    
+    for letter in word:
+        if str.upper(letter) in letter_bank:
+            temp_word += letter
+            word_str += str.upper(letter)
+    for letter in letter_bank:
+        if letter in letter_bank_dic:
+            letter_bank_dic[str.upper(letter)] += 1
+        else:
+            letter_bank_dic[str.upper(letter)] = 1
+    for letter in word_str:
+        if letter in word_dic:
+            word_dic[str.upper(letter)] += 1 
+        else:
+            word_dic[str.upper(letter)] = 1
+    if word_dic[letter] > letter_bank_dic[letter]:
+        return False
+    elif temp_word != word:
+        return False
+    elif temp_word == word:
+        return True   
```

Comment:
I see exactly what y'all were doing here. The problem is that you are making a variety of extra dictionaries and strings that just aren't necessary. In cases like this, it would be much better to make a deep copy of the letter_bank. From there, you can simply manipulate your copy.

The process would be as follows:

1) Make a deep copy of the letter bank. 
2) Go through each letter in the word and see if it is in the letter bank copy. 
     a) If it is, remove it from the letter bank copy.
     b) If it isn't return False
3) If you make it through the word without returning False, return True. 

This will take into account multiples of the same letter!